supabase/gotrue
# feat: add support for Twilio Verify

## What kind of change does this PR introduce?

Aims to add Twilio Verify Support. Twilio Verify is implemented as a separate provider. Only one of Twilio Verify or Twilio Programmable messaging an be selected. At this time, we only support the use of the `whatsapp` and `sms` channels with Twilio Verify.

This will affect the:
1. Signup flow
2. Verification flow (sms and phone_change)
3. Resend

The token is still generated, but not used in the Twilio Verify flow. It is used as a placeholder so as to try to ensure that to the OTP returned by the Verify service can only be used with the corresponding flow it was generated for. 

## What is the current behaviour?

We support programmable messaging.

## What is the new behaviour?

Developer can toggle between using Twilio Programmable Messaging on all flows or Twilio Verify on all flows.

## Additional context

Manual tests:

Probably need to be conducted on both Phone Change and SMS OTP Verification:

- [x] Existing Programmable Messaging (SMS/WhatsApp) (Signup/Verify/PhoneChange)
- [x] Twilio Verify(SMS/WhatsApp) 
- [ ] Update Frontend to include Twilio Verify

Admin methods shouldn't need to be updated to send to Twilio Verify since admin methods don't require confirmation
